core/chains/evm/client/pool.go

```diff
@@ -172,6 +182,31 @@ func (p *Pool) nLiveNodes() (nLiveNodes int, blockNumber int64, totalDifficulty
 	return
 }
 
+func (p *Pool) switchToBestNode() {
+	bestNode := p.nodeSelector.Select()
+	for _, n := range p.nodes {
+		// If a node is alive the SubscribersCount is guaranteed to be at least 1 because of the aliveLoop subscription
+		if n.State() == NodeStateAlive && n != bestNode && n.SubscribersCount() > 1 {
+			p.logger.Infof("Switching to best node from %q to %q", n.String(), bestNode.String())
+			n.UnsubscribeAll()
```

Comment:
based on the name of the this func, i expected it to set a value of 'best node' or drop and recreate connections, not just drop connections.

where are connections re-established?

Reply:
When a method on the pool gets called the "best" node is automatically selected, you can see this here: https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/client/pool.go#L399. 


The problem is when you have [a subscription](https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/client/pool.go#L440), it is a long-lived connection and it will not switch until the RPC goes down and the pool will signal all clients by calling the `unsubscribeAll` method.  The service that uses the long-lived connection has to deal with the connection re-establishment you mentioned (note that we are not terminating the connection from the pool to the RPC, we are terminating the connections from the services to the pool).

Reply:
i see. what do you think of expanding the comment to be ~ 
`Terminate connections to other nodes. Services are responsible for reconnecting, which will be routed to the new best node. Only terminate connections with more than 1 subscription to account for the aliveLoop subscription`

another question: does UnscribeAll kill the aliveLoop subscription? if so, then there is an asymmetry -- if there were 2 subscribers than the aliveLoop will be terminated but if there is 1 then it will continue to run.